app/Http/Controllers/PostsController.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Post;
+use App\PostStatus;
+use App\User;
+use Illuminate\Http\Request;
+
+class PostsController extends Controller
+{
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $posts = Post::all();
+
+        return view('posts.index', ['posts' => $posts]);
+    }
+
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        $users                 = User::all();
+        $post_statuses         = PostStatus::all();
+        $users_options         = [];
+        $post_statuses_options = [];
+
+        foreach ($users as $user) {
+            $users_options[$user->id] = $user->name;
+        }
+
+        foreach ($post_statuses as $post_status) {
+            $post_statuses_options[$post_status->id] = $post_status::STATUSES[$post_status->status];
+        }
+
+        return view('posts.create_form', [
+            'users'         => $users_options,
+            'post_statuses' => $post_statuses_options
+        ]);
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $post            = new Post($request->all());
+        $post->author_id = $request->author_id;
+        $post->status_id = $request->status_id;
+        $post->save();
+        $status = 'New post was created. It\'s id: ' . $post->id;
+
+        return view('posts.status', ['status' => $status]);
+    }
+
+    /**
+     * Display the specified resource.
+     *
+     * @param int $id
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        return view('posts.details', ['post' => Post::find($id)]);
```

Comment:
Add handler for not-found exception and add "404 Post not Found" page.

Reply:
Done